packages/react-chat/src/utils/stylesheet.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-console */
+import { useEffect, useState } from 'react';
+
+import { Assistant } from '@/common';
+
+// used to add stylesheets dynamically, resolves when loaded
+export const addStyleSheetURL = async (url: string) => {
+  const link = document.createElement('link');
+  link.rel = 'stylesheet';
+  link.href = url;
+
+  const load = new Promise((resolve, reject) => {
+    link.onload = resolve;
+    link.onerror = reject;
+  });
```

Comment:
I don't think this is possible, but technically, the resource starts loading right when you set the `href` attribute. So, in case the resource is already cached, the order of `.href` and `.onload` definitions could matter. Again, I don't think that's possible, because of event loop, and even the time it would take to load the cached resource, but better to be safe than sorry.



```suggestion
const load = new Promise((resolve, reject) => {
    link.onload = resolve;
    link.onerror = reject;
  });
  
  link.href = url;
```